nx/lib/nx/defn/grad.ex

```diff
@@ -234,11 +234,14 @@ defmodule Nx.Defn.Grad do
     {elem(tuple, index), cache}
   end
 
-  defp grad(:select, [pred, on_true, on_false], _ans, g, cache) do
-    {d_on_true, cache} = to_grad(on_true, g, cache)
-    {d_on_false, cache} = to_grad(on_false, g, cache)
-    result = Nx.select(pred, d_on_true, d_on_false)
-    {result, cache}
+  defp grad(:select, [pred, on_true, on_false], ans, g, cache) do
+    gs = Nx.broadcast(g, ans)
+    zeros = Nx.broadcast(Expr.tensor(0.0), ans)
```

Comment:
Is there a practical difference between using `ans`, `on_true` and `on_false` here? I'd guess not since they are all the same shape

Reply:
I think since `select` broadcasts `on_true` and `on_false`, it could be different if the shapes of `on_true` and `on_false` differ, but if they we're successfully broadcasted in the result, they'd successfully broadcast here as well.